provider/provider.go

```diff
@@ -561,13 +561,13 @@ func (p *Provider) GetNodeWithCtx(ctx context.Context, address string, options *
 
 // GetApps returns a page of applications known at the specified height and staking status
 // empty ("") staking_status returns all apps, page < 1 returns the first page, per_page < 1 returns 10000 elements per page
-func (p *Provider) GetApps(options *GetAppsOptions) (*GetAppsOutput, error) {
-	return p.GetAppsWithCtx(context.Background(), options)
+func (p *Provider) GetApps(options *GetAppsOptions, headers http.Header) (*GetAppsOutput, error) {
```

Comment:
I think we should make the headers part of the options parameter.

Reply:
OK updated.